pkg/api/defaults_test.go

```diff
@@ -1208,139 +1208,6 @@ func TestSetVMSSDefaultsAndZones(t *testing.T) {
 
 }
 
-func TestAKSDockerEngineDistro(t *testing.T) {
```

Comment:
let's update the existing tests to test that GPU nodes get the "AKS" or "AKS-18.04" distro

Reply:
I think the point of this PR is: there are no longer any VM SKU -- Distro relationships. (Happy to add more tests generally, but I don't think we want to add that one.)

Reply:
I think what I meant was "test that no configurations get the AKSDockerEngine distro" since that is deprecated but still exists. Ie. we should never assign the AKSDockerEngine distro, especially for a scale or update scenario.